internal/engine/upper.go

```diff
@@ -511,10 +515,14 @@ func handleConfigsReloaded(
 		manifestNames[m.Name] = true
 	}
 
-	b := state.TiltfileState.CurrentBuild
```

Comment:
I wish github let you comment outside of changed lines...

Some things in this function that I think will be buggy with multiple Tiltfiles:
1. If an extension loads a Tiltignore, it'll replace the main Tiltfile's Tiltignore (but also maybe it looks like `state.Tiltignore` could be deleted?)
2. The GC that happens with `newDefOrder` looks like it'll be broken unless we give `m.Source` a different value per Tiltfile
3. `state.Features` and `state.ConfigFiles` should probably be additive across Tiltfiles (they're already additive in the `err != nil` case) (also I think we can remove `state.ConfigFiles`)
4. Maybe TelemetrySettings, VersionSettings, et al should also be additive, but that seems a lot hairier